generate_buildjl.jl

```diff
@@ -139,14 +139,45 @@ end
 
 
 
-if length(ARGS) != 3
-	@error("Usage: generate_buildjl.jl path/to/build_tarballs.jl <repo_name> <tag_name>")
+if length(ARGS) < 1 || length(ARGS) > 3
+	@error("Usage: generate_buildjl.jl path/to/build_tarballs.jl [<repo_name> <tag_name>]")
     exit(1)
 end
 
 build_tarballs_path = ARGS[1]
-repo_name = ARGS[2]
-tag_name = ARGS[3]
+@info "Build tarballs script: $(build_tarballs_path)"
+src_name = basename(dirname(build_tarballs_path))
+if 2 <= length(ARGS) <= 3
+    repo_name = ARGS[2]
+else
+    repo_name = "JuliaBinaryWrappers/$(src_name)_jll.jl"
+end
+@info "Repo name: $(repo_name)"
+
+if length(ARGS) == 3
+    tag_name = ARGS[3]
+else
+    ctx = Pkg.Types.Context()
+    # Force-update the registry here, since we may have pushed a new version recently
+    BinaryBuilder.update_registry(ctx)
+    versions = VersionNumber[]
+    if any(isfile(joinpath(p, "Package.toml")) for p in Pkg.Operations.registered_paths(ctx.env, BinaryBuilder.jll_uuid("$(src_name)_jll")))
```

Comment:
This reads a little cleaner to me:
```suggestion
    paths = Pkg.Operations.registered_paths(ctx.env, BinaryBuilder.jll_uuid("$(src_name)_jll"))
    if any(p -> isfile(joinpath(p, "Package.toml")), paths)
```
Then you can reuse `paths` in the `for path in ...` loop below.

Reply:
Cool, thanks!